addons/mail/controllers/discuss.py

```diff
@@ -99,6 +99,8 @@ def _response_discuss_channel_invitation(self, channel_sudo, is_channel_token_se
                         'shouldDisplayWelcomeViewInitially': True,
                     })
                 channel_sudo = channel_sudo.with_context(guest=guest)
+                if channel_sudo.public == 'groups':
+                    raise NotFound
```

Comment:
Thank you for the update.

I would move the check above the first `guest =`, it is easier to read if the check is done upfront rather than at the end of the code (which in that case relies on the transaction rollback to cancel what was done above, instead of just not doing it).